python/ray/data/dataset.py

```diff
@@ -1319,10 +1195,9 @@ def std(self, on: Optional["AggregateOnTs"] = None, ddof: int = 1) -> U:
             return self._aggregate_result(ret)
 
     def sort(self,
-             key: Union[None, str, List[str], Callable[[T], Any]] = None,
+             key: Union[KeyFn, List[KeyFn]] = None,
```

Comment:
should we change the key type to just KeyFn, since we don't support multi keys yet and we haven't finalized on the syntax List[KeyFn] or List[(KeyFn, Bool)]

Reply:
Done